realm/realm-jni/src/io_realm_internal_LinkView.cpp

```diff
@@ -248,3 +250,15 @@ JNIEXPORT void JNICALL Java_io_realm_internal_LinkView_nativeRemoveTargetRow
         return lvr->remove_target_row( S(pos) );
     } CATCH_STD()
 }
+
+static void finalize_link_view(jlong ptr)
+{
+    TR_ENTER_PTR(ptr)
+    LangBindHelper::unbind_linklist_ptr(*LV(ptr));
+}
+
+JNIEXPORT jlong JNICALL Java_io_realm_internal_LinkView_nativeGetFinalizer
+  (JNIEnv *, jclass)
+{
+    return static_cast<jlong>(reinterpret_cast<uintptr_t>(&finalize_link_view));
```

Comment:
I had a discussion with @kneth before... actually this more readable than macro. So I think it is better we keep this style.